x-pack/test/apm_api_integration/tests/alerts/preview_chart_error_rate.spec.ts

```diff
@@ -0,0 +1,292 @@
+            },
+          },
+        };
+
+        const response = await apmApiClient.readUser({
+          endpoint: 'GET /internal/apm/rule_types/transaction_error_rate/chart_preview',
+          ...options,
+        });
+
+        expect(response.status).to.be(200);
+        expect(response.body.errorRateChartPreview).to.eql([]);
+      });
+
+      it('with no group by parameter', async () => {
+        const options = getOptions();
+        const response = await apmApiClient.readUser({
+          ...options,
+          endpoint: 'GET /internal/apm/rule_types/transaction_error_rate/chart_preview',
+        });
+
+        expect(response.status).to.be(200);
+        expect(response.body.errorRateChartPreview.length).to.equal(1);
+        expect(
+          response.body.errorRateChartPreview.map(
+            (item: { name: string; data: Array<{ x: number; y: number | null }> }) => item.name
+          )
+        ).to.eql(['synth-go_production_request']);
+      });
+
+      it('with default group by fields', async () => {
+        const options = {
+          params: {
+            query: {
+              ...getOptions().params.query,
+              groupBy: [SERVICE_NAME, SERVICE_ENVIRONMENT, TRANSACTION_TYPE],
+            },
+          },
+        };
+
+        const response = await apmApiClient.readUser({
+          ...options,
+          endpoint: 'GET /internal/apm/rule_types/transaction_error_rate/chart_preview',
+        });
+
+        expect(response.status).to.be(200);
+        expect(response.body.errorRateChartPreview.length).to.equal(1);
+        expect(
+          response.body.errorRateChartPreview.map(
+            (item: { name: string; data: Array<{ x: number; y: number | null }> }) => item.name
+          )
+        ).to.eql(['synth-go_production_request']);
+      });
+
+      it('with group by on transaction name', async () => {
+        const options = {
+          params: {
+            query: {
+              ...getOptions().params.query,
+              groupBy: [SERVICE_NAME, SERVICE_ENVIRONMENT, TRANSACTION_TYPE, TRANSACTION_NAME],
+            },
+          },
+        };
+
+        const response = await apmApiClient.readUser({
+          ...options,
+          endpoint: 'GET /internal/apm/rule_types/transaction_error_rate/chart_preview',
+        });
+
+        expect(response.status).to.be(200);
+        expect(response.body.errorRateChartPreview.length).to.equal(2);
+        expect(response.body.errorRateChartPreview[0].name).to.eql(
+          'synth-go_production_request_GET /apple 🍎 '
+        );
+        expect(response.body.errorRateChartPreview[1].name).to.eql(
+          'synth-go_production_request_GET /banana 🍌'
+        );
+        expect(response.body.errorRateChartPreview[0].data[0]).to.eql({
+          x: 1609459200000,
+          y: 25,
+        });
+        expect(response.body.errorRateChartPreview[1].data[0]).to.eql({
+          x: 1609459200000,
+          y: 50,
+        });
```

Comment:
Perhaps for readability:

```suggestion
        const appleTxPreview = response.body.errorRateChartPreview[0];
        const bananaTxPreview = response.body.errorRateChartPreview[1];
        
        expect(response.body.errorRateChartPreview.length).to.equal(2);
        expect(appleTxPreview.name).to.eql(
          'synth-go_production_request_GET /apple 🍎 '
        );
        expect(bananaTxPreview.name).to.eql(
          'synth-go_production_request_GET /banana 🍌'
        );
        expect(appleTxPreview.data[0]).to.eql({
          x: 1609459200000,
          y: 25,
        });
        expect(bananaTxPreview.data[0]).to.eql({
          x: 1609459200000,
          y: 50,
        });
```